sui/utils/upgrade-utils.js

```diff
@@ -51,9 +51,10 @@ async function upgradePackage(client, keypair, packageToUpgrade, contractConfig,
         arguments: [cap, receipt],
     });
 
+    const sender = options.sender || keypair.toSuiAddress();
```

Comment:
it is not. You can pass { signer: keypair } as an option to `signAndExecuteTransactions` and it will work, otherwise the sender has to be set (not sure why.